src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,379 @@
+        }
+
+
+        exitMessage();
+
+    }
+
+    public static void executeCommand(String input, Command c) throws InvalidCommandException, EmptyInputException, InvalidEventTimeException, InvalidDeadlineTimeException {
+        switch (c) {
+        case LIST:
+            listBeginMessage();
+            enumerateTasks();
+            break;
+        case DONE:
+            markTaskAsDone(input);
+            break;
+        case DELETE:
+            deleteTask(input);
+            decrementTasks();
+            break;
+        case TODO:
+            verifyValidInput(input, c);
+            addNewTodo(input);
+            confirmNewTask();
+            incrementTasks();
+            break;
+        case EVENT:
+            verifyValidInput(input, c);
+            addNewEvent(input);
+            confirmNewTask();
+            incrementTasks();
+            break;
+        case DEADLINE:
+            verifyValidInput(input, c);
+            addNewDeadline(input);
+            confirmNewTask();
+            incrementTasks();
+            break;
+        case INVALID:
+            throw new InvalidCommandException();
+
+        }
+    }
+
+    private static void loadData(String filepath) throws IOException {
+        File f = new File(filepath);
+        if (f.createNewFile()) {
+            System.out.println("Welcome to Duke. Is this your first time using Duke on this machine?");
+        } else {
+            System.out.println("Your previous Task list from Duke has been loaded! :-)");
+            processFileContents(filepath);
+        }
+    }
+
+    private static void saveData() {
+        try {
+            String data = getAllTaskListData();
+            writeToFile(filepath, data);
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        } catch (NullPointerException e) {
+            System.out.println("No items to add to file");
+        }
+    }
+
+    private static void processFileContents(String filePath) throws FileNotFoundException {
+        File f = new File(filePath); // create a File for the given file path
+        Scanner s = new Scanner(f); // create a Scanner using the File as the source
+        while (s.hasNext()) {
+            String line = s.nextLine();
+            processFileInputLines(line);
+        }
+    }
+
+    private static void processFileInputLines(String fileLine) {
+        char taskType = getTaskType(fileLine);
+        char isDone = getDone(fileLine);
+
+        incrementTasks();
+        addNewTaskFromFile(fileLine, taskType);
+        markCompletedTaskFromFile(isDone);
+
+    }
+
+    private static void markCompletedTaskFromFile(char isDone) {
+        if (isDone == 'Y') {
+            tasks.get(maxTaskIndex-1).markAsDone();
+        }
+    }
+
+    private static void addNewTaskFromFile(String fileLine, char taskType) {
+        String description;
+        switch (taskType) {
+        case ('T'):
```

Comment:
Naming the switch cases as a single letter makes it more ambiguous as to what the method is doing internally. Consider other naming possibilities that will improve readability.